src/xenia/kernel/xthread.cc

```diff
@@ -124,11 +137,15 @@ void XThread::set_last_error(uint32_t error_code) {
 }
 
 void XThread::set_name(const std::string& name) {
-  name_ = name;
+  StringBuffer buff;
```

Comment:
StringBuffer is expensive - it allocs like 16KiB by default -- there's `xe:: format_string("foo%s", ...)` that'll return an std::string you can use without to many extra copies/allocations.